### PR TITLE
Refactor adding the common fields to single method

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
To bring the logic of adding all the common fields into one method.

Issue # 

---

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/azure/walinuxagent/1707)

### CODE REVIEW

File: azurelinuxagent/common/future.py [hunk 1]
```diff
@@ -10,6 +10,11 @@
 except Exception:
     pass
 
+try:
+    from collections import OrderedDict  # For Py 2.7+
```
[the remark this answers is not in the document]
Comment (vrdmr) [reply azurelinuxagent/common/future.py]: Moved the whole section into the Py2 section check.

File: azurelinuxagent/common/future.py [hunk 2]
```diff
@@ -32,6 +34,10 @@
 
     bytebuffer = buffer
 
+    try:
```
Comment (narrieta) [inline azurelinuxagent/common/future.py]:
```
if sys.version_info[1] >= 7:
    from collections import OrderedDict
else:
    from ordereddict import OrderedDict

```

File: azurelinuxagent/common/event.py [hunk 3]
```diff
@@ -334,34 +335,59 @@ def add_metric(self, category, counter, instance, value, log_event=False):
...
-                              ('EventTid', threading.current_thread().ident),
-                              ('EventPid', os.getpid()), ("TaskName", threading.current_thread().getName()),
-                              ("KeywordName", '')]
+        ContainerId can change due to live migration and we want to preserve the container Id of the container writing
+        the event, rather than sending the event.
+        OpcodeName - This is used as the actual time of event generation.
+
+        :param event_parameters: List of parameters of the event.
+        :param set_values_for_agent: Need default values populated or not. Extensions need only GAVersion and
+                                            ContainerId to be populated and others sould be
```
Comment (larohra) [inline azurelinuxagent/common/event.py]: `others sould be` - NIT: `Should be` and I think the sentence is incomplete.

File: azurelinuxagent/common/event.py [hunk 4]
```diff
@@ -334,34 +335,59 @@ def add_metric(self, category, counter, instance, value, log_event=False):
...
+        :return: Event with default parameters populated (either values for agent or extension)
+        """
+        DefaultParameter = namedtuple('DefaultParameter', ['name', 'value'])
+        default_parameters = [DefaultParameter("GAVersion", CURRENT_AGENT),
+                              DefaultParameter('ContainerId', get_container_id_from_env()),
+                              DefaultParameter('OpcodeName', datetime.utcnow().__str__() if set_values_for_agent else ""),
```
Comment (larohra) [inline azurelinuxagent/common/event.py]: Can't we fill this up in general with the timestamp and overwrite it with extension data if they have something else in their event?

File: tests/common/test_event.py [hunk 5]
```diff
@@ -453,3 +454,131 @@ def test_report_metric(self, mock_event):
...
+                                          TelemetryEventParam('OpcodeName', '')]
+
+        extension_param_list_with_defaults = EventLogger.add_default_parameters_to_event(
+            extension_param_list_populated, set_values_for_agent=False)
+
+        # Converting list of TelemetryEventParam into a dictionary, for easier look up of values.
+        param_list_dict = OrderedDict([(param.name, param.value) for param in extension_param_list_with_defaults])
+
+        counter = 0
+        for p in default_parameters_expected_names:
```
Comment (larohra) [inline tests/common/test_event.py]: I see this logic being used in multiple places. Why not make this whole into a helper private function and just call that for better code cleanup and readability?

File: azurelinuxagent/common/event.py (hunk 4 above)
Comment (vrdmr) [reply azurelinuxagent/common/event.py]: For context: https://github.com/Azure/WALinuxAgent/pull/1663#discussion_r343233187

File: tests/common/test_event.py (hunk 5 above)
Comment (vrdmr) [reply tests/common/test_event.py]: Done.

Comment (vrdmr) [reply tests/common/test_event.py]: Done.

**Pull request merged**